sql/parse/parse.go

```diff
@@ -847,18 +849,25 @@ func convertAlterTable(ctx *sql.Context, ddl *sqlparser.DDL) (sql.Node, error) {
 		}
 		switch strings.ToLower(ddl.ConstraintAction) {
 		case sqlparser.AddStr:
-			if fkConstraint, ok := parsedConstraint.(*sql.ForeignKeyConstraint); ok {
+			switch c := parsedConstraint.(type) {
+			case *sql.ForeignKeyConstraint:
 				return plan.NewAlterAddForeignKey(
 					table,
-					plan.NewUnresolvedTable(fkConstraint.ReferencedTable, ddl.Table.Qualifier.String()),
-					fkConstraint), nil
-			} else {
+					plan.NewUnresolvedTable(c.ReferencedTable, ddl.Table.Qualifier.String()),
+					c), nil
+
+			case *sql.CheckConstraint:
+				return plan.NewAlterAddCheck(table, c), nil
+			default:
 				return nil, ErrUnsupportedFeature.New(sqlparser.String(ddl))
+
 			}
 		case sqlparser.DropStr:
 			switch c := parsedConstraint.(type) {
 			case *sql.ForeignKeyConstraint:
 				return plan.NewAlterDropForeignKey(table, c), nil
+			case *sql.CheckConstraint:
+				return plan.NewAlterDropCheck(table, c), nil
 			case namedConstraint:
```

Comment:
Add a TODO here: this is broken if they try to drop a check constraint without saying it's a check constraint

Reply:
added TODO, should `plan.NewAlterDropConstraint` just be generalized or do we need two different plans?